src/cpp/session/modules/SessionCodeTools.R

```diff
@@ -2587,3 +2587,31 @@
    getNamespace(envir)
    
 })
+
+.rs.addJsonRpcHandler("project_guess_cpp_style", function(inDirectory = .rs.getProjectDirectory()) 
+{
+   # give up if this is not a project or not a package
+   if (is.null(inDirectory)) return(.rs.scalar(""))
+
+   DESCRIPTION <- file.path(inDirectory, "DESCRIPTION")
+   if (!file.exists(DESCRIPTION)) return(.rs.scalar(""))
+
+   # check if Rcpp or cpp11 is mentioned in LinkingTo:
+   data <- read.dcf(DESCRIPTION)[1, ]
```

Comment:
You might want to guard against `read.dcf()` failures here, if (for some reason) the file was corrupt.

You can also use `read.dcf(DESCRIPTION, all = TRUE)` to get a `data.frame` back, which is typically easier to work with.